### PR TITLE
Fix button new pr

### PR DESCRIPTION
**Pull request by miladini**
This is the same as #26  after rebasing against the `main` branch.

It's been very long time since I coded in javascript, in case my array manipulation is not optimal, please leave a comment in the review.
Also, I tested this logic against this commit: 876c687b0819c18b2852e0edbcb14025348858f2 and not the current tip of main: 703d6d1db7d2dd2c62862f1da2a942c82ad874ee. And when tested, everything worked as expected.

Could one of you please test this if you get a chance?

@callmearta 
@eamirgh

### CODE REVIEW

**Comment by miladini** [comment]
Oh, I didn't realize this commit 9b3484dae47cc457bac9e02ac66c0be1f911a26c was new! It seems like you @callmearta have attempted to fix the follow button. I haven't tested this version, but if you already tested it, then we can close this PR.

Does this work though? I was wondering where `following_ids` is set for the first time?

**Comment by miladini** [comment]
I just tested this and it works as expected.

File: app/views/user.js [hunk 1]
```diff
@@ -146,11 +182,20 @@ const User = {
...
+                store.set('userData', userData);
+                this.$set(
+                        this.user,
+                        'user_profile',
+                        {
+                            ...this.user.user_profile,
+                            notification_type:null,
+                            my_following: false});
 				}
 			} else {
```
Comment (miladini) [inline app/views/user.js]: I just recalled before I tested this PR, I fixed a syntax error in here. There is a curly brace that is missplaced. I will correct it as soon as I can

**Review by miladini — COMMENTED** [review]
:

File: app/views/user.js (hunk 1 above)
Comment (miladini) [reply app/views/user.js]: done

**Comment by callmearta** [comment]
What's the main purpose of this PR mate? If it's fixing the follow button i've already fixed it in latest version.

**Comment by miladini** [comment]
I see. I missed that last commit. Let's close this PR then.

**Pull request closed without merging**